src/vaadin-grid-data-provider-mixin.html

```diff
@@ -231,10 +231,14 @@
     }
 
     _isExpanded(item) {
-      return this.expandedItems && this._getItemIndexInArray(item, this.expandedItems) > -1;
+      return this.__expandedKeys.has(this.getItemId(item));
     }
 
     _expandedItemsChanged(e) {
+      this.__expandedKeys = new Set();
+      this.expandedItems.forEach(item => {
+        this.__expandedKeys.add(this.getItemId(item));
```

Comment:
If `itemidPath` is changed dynamically, the `__expandedKeys` would expire, but it wouldn't get repopulated. Need to add `itemIdPath` to the observer properties

Reply:
Created observer that updates `__expandedKeys` if `itemIdPath` changes.